### DISCUSSION TITLE
Is there a way to create custom scripts

### DISCUSSION THREAD

**Discussion by Estegar** [usebruno/bruno#1366] [General]
I'm trying out Bruno and so far it looks promising, we want to switch from Postman. The only thing i need here is a way to make some helper functions, that i will later import/require in my test scripts.
I want to create a .js file with some functions/classes etc. and then require them in test scripts?

Is there a way to do it?
Maybe there is a place where i can define my functions that will be available in test scripts?
I'm not very familiar with javascript, maybe i'm missing something here?

**Comment by Estegar** [comment]
Nevermind, i just need to create new javascript module in js like normal and it works without problems :P
For anyone wondering how to do it here's a sample:

```
function helloWorld() {
    console.log("Hello world")
}

exports.helloWorld = helloWorld
```


you put that file in node_modules folder with name helloWorld.js then you do this in Bruno:

```
const { helloWorld } = require('helloWorld')

helloWorld()
```

you should see in console "Hello world" message.

**Comment by helloanoop** [comment]
Nice that you figured it out @Estegar !

I would not recommend putting it inside `node_modules` as it is reserved to packages installed by `npm`

I would recommend creating a `lib` or `helpers` folder to do this.
I demoed it in the [youtube livestream](https://www.youtube.com/live/ZtNGbglTjps?si=z9RlH4R_voFFAjpp&t=4974) checkout 3 mins at 1h:23 mark in the video

**Comment by varunvora** [comment]
Can we have the custom script shared between collections? I have a common auth script for all collections but do not want to duplicate it.

Edit: Found a solution! For anyone looking to share scripts between collections: https://github.com/usebruno/bruno/discussions/2000#discussioncomment-9001150
